rfd/xxxx-bot-for-cert-renewals.md

```diff
@@ -0,0 +1,626 @@
+- a new `teleport.dev/owned-by` label is applied to all resources related to a
+  particular bot
+
+Note: end users should never need to manually create, modify, or delete these
+objects. All operations on bots and their related objects is performed through
+the `tctl bots` subcommand.
+
+#### `tbot` configuration
+
+The goal is to keep tbot configuration to a minimum to make for a quick getting
+started experience, however we recognize many use cases may quickly become
+difficult to encode on the command line.
+
+To that end, we'd like to follow the philosophy of "keep simple things simple
+but make complex things possible": the simplest / most popular use cases should
+ideally require minimal (zero?) configuration beyond a concise CLI, but more
+complicated use cases can make use of a `tbot.yaml` configuration file.
+
+An example `tbot.yaml` with all advanced options specified:
+
+```yaml
+# Storage for the bot's private data
+storage:
+  # can also support any backend supporting both store + fetch
+  directory: /var/lib/tbot
+
+sinks:
+  - directory: /home/alice/.tbot
+    roles: [a, b, c]
+
+  # TODO: probably need to specify permissions (this will be ugly)
+  - directory: /home/bob/.tbot
+    roles: [b, c]
+
+  # k8s secrets support store and fetch
+  - kubernetesSecret:
+      kubeconfig: /home/user/kubeconfig
+      namespace: example
+      secretName: example-secret
+    roles: [c]
+
+  # Webhooks support only store
+  - webhook: http://localhost:8001
+    roles: [a]
+```
+
+#### External Configuration Assist
+
+In order to make it easy to consume certificates generated by `tbot`, the agent
+will have a `tbot config` subcommand that can render configuration snippets
+for common integration points.
+
+For example:
+
+```
+$ tbot start ...
+$ tbot config --ssh
+To configure sshd to trust this host certificate, add the following to your sshd_config:
+
+HostCertificate /foo/bar/tbot-cert.pub
+```
+
+(In the above, the instructional text will be written to stderr, so that the following
+will be possible: `tbot config --ssh | sudo tee -a /etc/ssh/sshd_config`)
+
+If `tbot` is managing multiple `--cert`s, you can pass `mode:destination`
+to tell it which certificate to render configuration for.
+
+Note: `tctl auth sign` has a tiny amount of configuration templating already for
+Postgres and MongoDB certificates. We should aim to move this logic out into a
+separate package that can render configuration snippets for a variety of
+external systems. This work is out of scope for the purposes of this RFD.
+
+#### Polling for expiration
+
+In order to watch a certificate for expiration, the
+`teleport/lib/utils/interval` package will be used. Note that tbot will only set
+up polling for _user certs_, as host certificates issued by teleport do not
+expire (host certificates *will* be re-issued during CA rotations).
+
+#### Initial User Certificates
+
+In order for the bot to obtain its first set of user certificates, a new
```

Comment:
How would this work with IAM node joining methods?

Reply:
I'll add a small note to this section about accepting EC2 identity documents.